PowerEditor/src/ScintillaComponent/columnEditor.cpp

```diff
@@ -57,7 +57,15 @@ intptr_t CALLBACK ColumnEditorDlg::run_dlgProc(UINT message, WPARAM wParam, LPAR
 			::SendDlgItemMessage(_hSelf, IDC_COL_LEADING_COMBO, CB_ADDSTRING, 0, reinterpret_cast<LPARAM>(TEXT("None")));
 			::SendDlgItemMessage(_hSelf, IDC_COL_LEADING_COMBO, CB_ADDSTRING, 0, reinterpret_cast<LPARAM>(TEXT("Zeros")));
 			::SendDlgItemMessage(_hSelf, IDC_COL_LEADING_COMBO, CB_ADDSTRING, 0, reinterpret_cast<LPARAM>(TEXT("Spaces")));
-			::SendMessage(::GetDlgItem(_hSelf, IDC_COL_LEADING_COMBO), CB_SETCURSEL, colEditParam._leadingChoice, 0);
+			WPARAM curSel = 0;

```

Comment:
May I ask why this block is necessary?
Can't `leadingChoice` be `int`?

Reply:
Don wanted the "leading choice" to be an enum (see Don's prior comments).  My goal here was to fully uncouple things.  Again, I can't read Don's mind as to how "far" to take such things, so I proceeded on the presumption of this type of decoupling being wanted.  I really have no idea.  If the future gets "sticky" for me on this PR, I will abandon it and close it; of course an outright rejection will do that just as well, also.  :-)

Reply:
> Don wanted the "leading choice" to be an enum

I meant to ask why the enum couldn't be `int`. :)

Well, let's for Don's review.

Reply:
Yea, the way I read it was Don specifically wanted `enum`.
But maybe his real intent was to just not have two bool.  I do understand the desire to not have two bool; indeed I only went with 2 bool in the beginning in an effort to change as little as possible.
The high volume of changed lines in this makes me uncomfortable -- not at a C++ level, but at a "this is someone else's project" level.
I see others aren't afraid to change a lot of lines; I think a lot of changes are more acceptable if the feature is really wanted.
I'm truly sorry that I got involved in this one; my intent was to stay with smaller changes.

Reply:
I'm glad you got involved in this one. - You did a great job. 👍
I'm sorry you didn't enjoy working on it.

Reply:
@alankilborn 
Yes, I did want an `enum` but not a `int`.
Regarding the quantity of modification, it depends on different context and a lot of factors, so there is no really a standard criteria.
But just keep one thing in mind - if the a function could have 3 lines but it has 20 lines - then the function is usually coded badly.
Also the unnecessary diff (reformatting the code or it could be generated badly by the tool) will be rejected for sure - the goal is not only to make code review easier, but also for the maintenance (it happens that we go back old commits for debugging reason).  
OK, I will do the enum myself after merging this PR, if you are not willing do it

Reply:
@Yaron10 
> I meant to ask why the enum couldn't be int. :)

Because specific typed variable is better than a generic typed variable to make the function usage more explicit. Don't you think so?

Reply:
@donho,

My main question was why the `switch (colEditParam._leadingChoice)` block was necessary.

In the first commit here, only `::SendMessage(::GetDlgItem(_hSelf, IDC_COL_LEADING_COMBO), CB_SETCURSEL...)` was called.

And this is relevant to `ColumnEditorDlg::getLeading()` as well.

Reply:
>My main question was why the switch (colEditParam._leadingChoice) block was necessary.

"Necessary" might be a strong word, here. :-)
We can do things in a "hacky" way, and sure, that has been done in the past.

The code (that I think) you are questioning was changed to fully decouple the integer values of the combobox selection indices and the "values" of the `enum` type.

With this, the code for this line:
`enum leadingChoice : UCHAR { noneLeading, zeroLeading, spaceLeading };`
could be changed to this without any runtime difference:
`enum leadingChoice : UCHAR { noneLeading=27, zeroLeading=45, spaceLeading=106 };`

(Probably the `UCHAR` should be eliminated; this is a carryover of my habit because I work on smallish embedded systems where every byte can matter.)

Don seemed to want an `enum` ; when he requested this change, I provided, in a better and more complete fashion than an earlier commit (earlier commits aren't really relevant when changes are requested and made).

Reply:
@donho 

>Edit: oups, I misread your comment. Good job, it's what it needs.

Thank you, Don, encouragement to your contributors that they are doing the right thing is very helpful indeed!

Reply:
@alankilborn,

Thank you for the explanation. 👍 
You're a very good coder, and I was trying to understand and learn your approach.